Yoast/wordpress-seo
# Passed posts' main image url to new mobile snippet preview

## Summary

Post/ Pages:
* If a featured image has been set, the image url will be passed to the mobile snippet preview.
* If a featured image has not been set, the url of the first image in the content will be passed.
* If the post does not contain any images, `null` will be returned.

Taxonomies:
* If the term contains an image, the image url will be passed to the mobile snippet preview.
* If the term does not contain any images, `null` will be returned.


This PR can be summarized in the following changelog entry:

* Adds an image to the new mobile snippet preview for posts and terms.

## Relevant technical choices:

* Created a function which gets the post image url (featured image, else first image in post, else returns `null`)
* Created a function which gets the term description's image url (first image in term description, else returns `null`)
* Created a function which gets the first image of a term.
* Added the `socialPreviewImageURL` to the `PostScraper` and `TermScraper` store.
* Mapped the `socialPreviewImageURL` (as `mobileImageSrc`) to the snippet editor component.
* Wrote unit tests

## Test instructions

This PR can be tested by following these steps:

* Add an image in a posts' content, save and reload the page.
    * Check if the image shows up in the mobile snippet preview.
<img src="https://user-images.githubusercontent.com/42736463/64341480-bb1b4680-cfe8-11e9-80d7-8d838891de71.png" width="200" height="360" />

* Add a featured image to the post, save and reload the page.
     * Check whether the featured image shows up in the mobile snippet preview, replacing the posts' first image.
![image](https://user-images.githubusercontent.com/42736463/64341345-7099ca00-cfe8-11e9-8b7a-5b294e146ffc.png)

* Remove the image in the content of the post, save and reload the page.
    * Check if the featured image still shows up in the mobile snippet preview.

* Remove the featured image, save and reload the page.
    * Check if the image/ placeholder in the mobile snippet preview disappears completely.
![image](https://user-images.githubusercontent.com/42736463/64341769-5b716b00-cfe9-11e9-802f-a33e32fef827.png)

Also check this for:
* Page
* Tag
* Category

## UI changes

## Documentation

## Quality assurance


Fixes #13058
